### DISCUSSION TITLE
robust optimization

### DISCUSSION THREAD

**Discussion by lolloame** [e0404/matRad#591] [Q&A] [answered]
Hi, I would like to know which kind of robust optimization is runned in matRad, if the probabilistic one ore the worst-case one.
Thank you in advance.

**Comment by wahln** [comment, marked as answer]
Robust optimization is implemented on the dev_varRBErobOpt branch. It has multiple techniques implemented which can be set using the robustness property of objectives. The techniques are
- STOCH: expected value minimization (probably what you mean with the probabilistic approach).
- COWC: composite worst case, which is the worst case of the sum of all COWC objectives
- OWC: objective worst case, which minimizes the individual worst case of each objective
- VWWC: voxel-wise worst case, which defines the worst case for each voxel as the highest dose value among all scenarios and uses this dose distribution for evaluating the objective
- VWWC_INV: similar to VWWC, only that the lowest dose values is defined as the worst-case

For COWC & OWC usually use a minimax approach in which the maximum is approximated by logsumexp to allow differentiation.

Note that matRad separates between uncertainty model (Gaussian), scenario model (random or gridded shifts) and optimization.
This can be confusing: For example, you have a gaussian shift model with +-2mm for which you can define a worst case (in multiples of sigma). 
You can then choose a worst case scenario model that will create shifts according to this worst case definition, or a scenario model that randomly samples from the multivariate Gaussian. 
After this, you can again decide on the optimization type. So in principle, you can use a random model with a COWC optimization, or an individual shift/worst-case model with the STOCH optimization.

Since the branch is still in preparation and not yet released, we currently don't have any more documentation on this, but I hope my input helps you navigate the code.

**Reply by lolloame** [reply]
Thanks a lot @wahln, you've been very helpful.
